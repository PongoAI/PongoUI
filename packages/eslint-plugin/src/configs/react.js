/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
