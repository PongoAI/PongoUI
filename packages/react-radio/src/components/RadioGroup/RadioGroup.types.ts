import type {
  RadioGroupProps as FluentRadioGroupProps,
  RadioGroupState as FluentRadioGroupState,
} from '@fluentui/react-radio';

export type RadioGroupProps = FluentRadioGroupProps;

export type RadioGroupState = FluentRadioGroupState;
