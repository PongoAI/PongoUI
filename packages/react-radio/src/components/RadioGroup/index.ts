export * from './RadioGroup';
export * from './useRadioGroupStyles';
