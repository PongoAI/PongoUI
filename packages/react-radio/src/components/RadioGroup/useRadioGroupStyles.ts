import { mergeClasses, makeStyles, shorthands } from '@griffel/react';
import type { RadioGroupState } from './RadioGroup.types';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    alignItems: 'flex-start',
    ...shorthands.gap('10px'),
  },

  vertical: {
    flexDirection: 'column',
  },
});

export const useRadioGroupStyles = (state: RadioGroupState): RadioGroupState => {
  const styles = useStyles();

  state.root.className = mergeClasses(
    styles.root,
    styles.root,
    state.layout === 'vertical' && styles.vertical,
    state.root.className,
  );

  return state;
};
