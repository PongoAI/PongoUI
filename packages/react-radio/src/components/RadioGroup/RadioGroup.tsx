import * as React from 'react';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
import { RadioGroupProps } from './RadioGroup.types';
import { useRadioGroup_unstable, renderRadioGroup_unstable } from '@fluentui/react-radio';
import { useRadioGroupStyles } from './useRadioGroupStyles';

/**
 * A RadioGroup component presents a set of options where only one option can be selected.
 */
export const RadioGroup: ForwardRefComponent<RadioGroupProps> = React.forwardRef((props, ref) => {
  const state = useRadioGroup_unstable(props, ref);

  useRadioGroupStyles(state);

  return renderRadioGroup_unstable(state);
});

RadioGroup.displayName = 'RadioGroup';
