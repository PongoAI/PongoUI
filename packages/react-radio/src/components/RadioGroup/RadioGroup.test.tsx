import React from 'react';
import { RadioGroup } from './RadioGroup';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('RadioGroup', () => {
  isConformant({
    Component: RadioGroup,
    displayName: 'RadioGroup',
    disabledTests: ['consistent-callback-args'],
  });

  it('renders correctly', () => {
    const result = render(<RadioGroup>Test</RadioGroup>);
    expect(result.container).toMatchSnapshot();
  });
});
