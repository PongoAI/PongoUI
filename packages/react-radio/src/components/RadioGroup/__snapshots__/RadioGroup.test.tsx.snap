// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders correctly 1`] = `
<div>
  <div
    class="___17ne5cp f22iagw f6jr5hl f1aaj51y ft97ctm f1vx9l62"
    role="radiogroup"
  >
    Test
  </div>
</div>
`;
